stack.c

```diff
@@ -3,57 +3,60 @@
 
 #include "stackconfig.h"
 #include "stack.h"
+void init(Stack* stack) {
+    stack->top = -1;
+}
+void push(Stack* stack, stackType input) {
+
+    if (isFull(stack)) {
+        {
+            printf("Stack overflow while push()");
+            return;
+        }
+        stack->top = stack->top + 1;
+        stack->data[stack->top] = input;
 
-void push(Stack* stack, stackType data) {
-    Node* newNode = malloc(sizeof(Node));
-    if(newNode == NULL) {
-        printf("Stack overflow while push()");
-        return;
+        //newNode->data = data;
+        //newNode->next = stack->top;
+        //stack->top = newNode;
     }
-    newNode->data = data;
-    newNode->next = stack->top;
-    stack->top = newNode;
 }
+    stackType pop(Stack * stack) {
+        if (isEmpty(stack)) {
+            printf("Stack underflow while pop\n");
+            return -1;
+        }
+        stackType a = stack->data[stack->top];
```

Comment:
avoid unnecessary changes like changing sensible variables names (data) to non-sensical ones (a)